### PR TITLE
chore: test displaying flashcards

### PR DESCRIPTION
**Pull request by mmcode9407**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
# Code Review Report

## Task goal
Write an integration test covering displaying flashcards

## Requirements
You need to cover following scenarios:
1. It should display flashcards in the list properly

## Code Review

The provided code changes seem to cover the task requirements. The test case `should get all flashcard` is checking if all flashcards are displayed correctly in the list. 

The test is mocking a response from a `getCardsRequest` function to return an array of flashcards. Then it renders the `App` component and checks if all flashcards from the mocked response are displayed on the screen.

The test case is correctly using `waitFor` function from `@testing-library/react` to wait for the asynchronous operation of fetching and displaying flashcards.

The test case is correctly using `getAllByText` function from `@testing-library/react` to find all elements that have a text content of 'front'. It then checks if the number of such elements is equal to the number of flashcards in the mocked response.

In conclusion, the code changes cover the task requirements and there are no apparent bugs in the code.
